### PR TITLE
RandomARQ hashing implementation

### PR DESCRIPTION
**Pull request by malbit**
blockchain: keep alternative blocks in blockchain file

Alternative blocks are cleared on startup unless --keep-alt-blocks is passed on the command line

Needed for New hashing algorithm called: RandomARQ based on RandomX

StageNet V15 ForkBlock is set to 12100

### CODE REVIEW

**Review by ArqTras — DISMISSED** [review]

**Review by ArqTras — DISMISSED** [review]
Working on my end and passed fork block

**Review by ArqTras — DISMISSED** [review]

**Review by ArqTras — DISMISSED** [review]
Devil's Randomness as community decide
No issues with msys build

**Review by ArqTras — DISMISSED** [review]

File: src/crypto/CMakeLists.txt [hunk 1]
```diff
@@ -80,6 +83,7 @@ arqma_add_library(cncrypto
 target_link_libraries(cncrypto
   PUBLIC
     epee
+    randomx
```
Comment (LithyRiolu) [inline src/crypto/CMakeLists.txt:86]: This should be `randomarq` otherwise it won't work as desired.

File: src/crypto/CMakeLists.txt [hunk 2]
```diff
@@ -48,8 +48,11 @@ set(crypto_sources
   random.c
   skein.c
   slow-hash.c
+  rx-slow-hash.c
```
Comment (LithyRiolu) [inline src/crypto/CMakeLists.txt:51]: Push this line up 2 lines to keep consistent alphabetical order.

**Review by LithyRiolu — CHANGES_REQUESTED** [review]
*sorry for just throwing my review over without any form of notice.*

* 2 lines need attention
* randomarq won't be initialized within `crypto` because of a typo
* everything else looks good as far as I can tell.

File: src/crypto/CMakeLists.txt (hunk 1 above)
Comment (malbit) [reply src/crypto/CMakeLists.txt:86]: Not really. Cus randomarq is doing randomx lib. It was tested at stagenet and working as it should

File: src/crypto/CMakeLists.txt (hunk 2 above)
Comment (malbit) [reply src/crypto/CMakeLists.txt:51]: It dont need to be in alphabetical order

Comment (TheDevMinerTV) [reply src/crypto/CMakeLists.txt]: 97571858675ffa631283be27f1d4ef6160fe9053

File: src/crypto/CMakeLists.txt (hunk 1 above)
Comment (TheDevMinerTV) [reply src/crypto/CMakeLists.txt:86]: But name interference I guess...

Comment (malbit) [reply src/crypto/CMakeLists.txt:86]: RandomARQ is based on RandomX

**Review by ArqTras — APPROVED** [review]

**Pull request merged**